File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        current_minimum = prices[0]
        current_profit = 0

        for i in range(1, len(prices)):
            if prices[i] > current_minimum:
                current_profit = max(current_profit, prices[i] - current_minimum)
                profit = max(profit, current_profit)
            else:
                current_minimum = prices[i]
        
        return profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        var buying = prices[0]
        var result = 0
        
        for price in prices {
            if price < buying {
                buying = price
            } else if result < (price - buying) {
                result = price - buying
            }
        }
        
        return result
    }
}